python_src/tests/performance_manager/test_performance_manager.py

```diff
@@ -174,64 +174,160 @@ def mock__get_static_parquet_paths(
         mock__get_static_parquet_paths,
     )
 
-    # pylint: disable=R0913
-    # pylint too many arguments (more than 5)
-    def mock__write_parquet_file(
-        table: Table,
-        file_type: str,
-        s3_dir: str,
-        partition_cols: List[str],
-        visitor_func: Optional[Callable[..., None]] = None,
-        basename_template: Optional[str] = None,
-    ) -> None:
+    yield
+
+
+@pytest.fixture(name="_flat_file_s3_patch")
+def fixture_flat_file_s3_patch(monkeypatch: MonkeyPatch) -> Iterator[None]:
+    """
+    monkeypatch s3 interactions for the flat file writing function
+    """
+    public_archive_file_details = [
+        {
+            "filepath": "s3://bucket_name/lamp/subway_on_time_performance-v1/index.csv",
+            "size": 123,
+            "last_modified": datetime.datetime.now(),
+        },
+        {
+            "filepath": "s3://bucket_name/lamp/subway_on_time_performance-v1/2023-04-06-subway-on-time-performance-v1.parquet",
+            "size": 123,
+            "last_modified": datetime.datetime.now(),
+        },
+        {
+            "filepath": "s3://bucket_name/lamp/subway_on_time_performance-v1/2023-04-07-subway-on-time-performance-v1.parquet",
+            "size": 123,
+            "last_modified": datetime.datetime.now(),
+        },
+    ]
+
+    test_archive_value = "TEST_GTFS_PUBLIC"
+    monkeypatch.setattr(S3Archive, "BUCKET_NAME", test_archive_value)
```

Comment:
i tried setting this as an env var, but it gets loaded before this monkey patch is applied.